controller/eventGroupController.js

```diff
@@ -0,0 +1,172 @@
+const HttpStatus = require("http-status-codes");
+
+const EventGroup = require("../model/eventGroupModel");
+const Event = require('../model/eventModel');
+const ValidationUtil = require('../util/validationUtil');
+const StatusEnum = require('../model/enums').StatusEnum;
+
+
+// =================================== CRUD FUNCTIONS ===================================
+
+
+const create = (req, res) => {
+    const loggedUser = req.decoded;
+
+    if (!loggedUser || !req.body) {
+        return res.status(HttpStatus.BAD_REQUEST).json({
+            status: StatusEnum['ERROR'],
+            message: 'Bad request'
+        });
+    }
+    // TODO: AdminAuth MW
+    // TODO: add new user to eventGroup (update eventGroup.users)
+    // TODO: remove user from eventGroup
+    const eventGroup = new EventGroup(req.body);
+
+    if (loggedUser.companyID != eventGroup.companyID) {
+        return res.status(HttpStatus.UNAUTHORIZED).json({
+            status: StatusEnum['ERROR'],
+            message: 'Unauthorized'
+        });
+    }
+
+    // Check if event exists 
+    Event.findOne({ _id: eventGroup.eventID, companyID: eventGroup.companyID }).then(event => {
+        if (!event) {
+            return res.status(HttpStatus.NOT_FOUND).json({
+                status: StatusEnum['ERROR'],
+                message: 'Event not found'
+            });
+        } else {
+            eventGroup.save().then(() => {
+                return res.status(HttpStatus.CREATED).json(eventGroup);
+            }).catch(err => {
+                return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+                    status: StatusEnum['ERROR'],
+                    message: ValidationUtil.buildErrorMessage(err, 'create', 'eventGroup')
+                });
+            });
+        }
+    }).catch(err => {
+        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+            status: StatusEnum['ERROR'],
+            message: ValidationUtil.buildErrorMessage(err, 'create', 'eventGroup')
+        });
+    });
+};
+
+const retrieveAll = (req, res) => {
```

Comment:
I do not understand this function. It seems it gets all of the event groups inside the company, even the ones the `loggedUser` is not part of. This should be changed as follows:
- If the `loggedUser` (maybe rename to `loggedInUser`, makes more sense gramatically) is of `accessType` Super-Admin/Admin then he can get all event groups
- Otherwise, if `loggedInUser` is of `accessType` Employee he can only get those event groups he is a part of